File: src/CSES.Solutions/Introductory/WeirdAlgorithm/WeirdAlgorithmTests.cs
using Xunit;

namespace CSES.Solutions.Introductory;

/// <summary>
/// Tests for the Weird Algorithm problem.
/// </summary>
public class WeirdAlgorithmTests : BaseSolverTests<WeirdAlgorithmSolver>
{
    private const string TestDataFolder = "Introductory/WeirdAlgorithm/TestData";

    [Theory]
    [InlineData(1)]
    [InlineData(2)]
    [InlineData(3)]
    public void Test(int testNumber)
    {
        RunTest(TestDataFolder, testNumber);
    }

    // Alternative: Use this instead if you want to auto-discover all test files
    // [Theory]
    // [MemberData(nameof(GetAllTests))]
    // public void Test(int testNumber)
    // {
    //     RunTest(TestDataFolder, testNumber);
    // }
    //
    // public static IEnumerable<object[]> GetAllTests()
    //     => GetTestNumbers(TestDataFolder);
}

File: src/CSES.FSharp.Solutions/IntroductoryProblems/WeirdAlgorithm/WeirdAlgorithmTests.fs
namespace CSES.FSharp.Solutions.IntroductoryProblems

open Xunit

/// Tests for the Weird Algorithm problem.
type WeirdAlgorithmTests() =
    inherit CSES.FSharp.Solutions.BaseSolverTests<WeirdAlgorithmSolver>()

    static let testDataFolder = "IntroductoryProblems/WeirdAlgorithm/TestData"

    [<Theory>]
    [<InlineData(1)>]
    [<InlineData(2)>]
    [<InlineData(3)>]
    [<InlineData(4)>]
    [<InlineData(5)>]
    [<InlineData(6)>]
    [<InlineData(7)>]
    [<InlineData(8)>]
    [<InlineData(9)>]
    [<InlineData(10)>]
    [<InlineData(11)>]
    [<InlineData(12)>]
    [<InlineData(13)>]
    [<InlineData(14)>]
    member this.Test(testNumber: int) =
        this.RunTest(testDataFolder, testNumber)
